### PR TITLE
find dapr.yaml in dirpath when dapr run -f <dirpath> is executed

### PR DESCRIPTION
**Pull request by pravinpushkar**
Signed-off-by: Pravin Pushkar <ppushkar@microsoft.com>

# Description

Please reference the issue this PR is related to: #1143

### CODE REVIEW

File: utils/utils.go [hunk 1]
```diff
@@ -352,3 +353,28 @@ func ReadFile(filePath string) ([]byte, error) {
 	}
 	return bytes, nil
 }
+
+// IsYAMLFile returns true if the given file path is a valid YAML file.
+func IsYAMLFile(filePath string, fs afero.Afero) bool {
```
Comment (shubham1172) [inline utils/utils.go:358]:
```suggestion
func IsYAMLFile(fs afero.Afero, filePath string) bool {
```

Comment (pravinpushkar) [reply utils/utils.go]: removed `afero` dep for now.

File: utils/utils.go [hunk 2]
```diff
@@ -352,3 +352,28 @@ func ReadFile(filePath string) ([]byte, error) {
...
+		return false
+	}
+	isDir := fileInfo.IsDir()
+	hasYAMLExtension := strings.HasSuffix(filePath, ".yaml") || strings.HasSuffix(filePath, ".yml")
+	return !isDir && hasYAMLExtension
+}
+
+// FindFileInDir finds and returns the path of the given file name in the given directory.
+func FindFileInDir(dirPath, fileName string) (string, error) {
+	files, err := os.ReadDir(filepath.Clean(dirPath))
```
Comment (mukundansundar) [inline utils/utils.go:369]: check if it is a dir first?

File: cmd/run.go [hunk 3]
```diff
@@ -62,6 +62,7 @@ var (
 )
 
 const (
+	defaultRunFileName          = "dapr.yaml"
```
Comment (mukundansundar) [inline cmd/run.go:65]:
```suggestion
	defaultRunTemplateFileName          = "dapr.yaml"
```

File: utils/utils.go [hunk 4]
```diff
@@ -352,3 +352,28 @@ func ReadFile(filePath string) ([]byte, error) {
...
+	if err != nil {
+		return "", fmt.Errorf("error reading directory %q : %w", dirPath, err)
+	}
+	for _, f := range files {
+		if !f.IsDir() && f.Name() == fileName {
+			return filepath.Join(dirPath, fileName), nil
+		}
+	}
```
Comment (mukundansundar) [inline utils/utils.go:377]: why list files and iterate through them? 
Join the dirPath and defaultRunTemplateFileName and see if such a file exists?

**Review by mukundansundar — CHANGES_REQUESTED** [review]

File: utils/utils.go (hunk 4 above)
Comment (pravinpushkar) [reply utils/utils.go:377]: aah..right.

File: cmd/run.go [hunk 5]
```diff
@@ -441,3 +447,27 @@ func executeRunWithAppsConfigFile(runFilePath string) {
...
+	fileInfo, err := os.Stat(path)
+	if err != nil {
+		return "", fmt.Errorf("error getting file info for %s: %w", path, err)
+	}
+	if fileInfo.IsDir() {
+		filePath, err := utils.FindFileInDir(path, defaultRunTemplateFileName)
+		if err != nil {
+			return "", err
+		}
+		fmt.Println("Using run file: ", filePath)
```
Comment (mukundansundar) [inline cmd/run.go:464]: remove

File: cmd/run.go [hunk 6]
```diff
@@ -441,3 +447,27 @@ func executeRunWithAppsConfigFile(runFilePath string) {
...
+		return filePath, nil
+	}
+	hasYAMLExtension := strings.HasSuffix(path, ".yaml") || strings.HasSuffix(path, ".yml")
+	if !hasYAMLExtension {
+		return "", fmt.Errorf("file %q is not a YAML file", path)
+	}
+	fmt.Println("Using run file file: ", path)
```
Comment (mukundansundar) [inline cmd/run.go:471]: remove

File: utils/utils_test.go [hunk 7]
```diff
@@ -267,6 +267,64 @@ func TestReadFile(t *testing.T) {
 	}
 }
```
Comment (mukundansundar) [inline utils/utils_test.go:269]: add a comment on the dir structure within the temp dir

**Review by shubham1172 — APPROVED** [review]

**Review by mukundansundar — APPROVED** [review]

**Pull request merged**